authorized_keys_command.sh

```diff
@@ -33,12 +33,23 @@ then
   export AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN AWS_SECURITY_TOKEN
 fi
 
-UnsaveUserName="$1"
-UnsaveUserName=${UnsaveUserName//".plus."/"+"}
-UnsaveUserName=${UnsaveUserName//".equal."/"="}
-UnsaveUserName=${UnsaveUserName//".comma."/","}
-UnsaveUserName=${UnsaveUserName//".at."/"@"}
-
-aws iam list-ssh-public-keys --user-name "$UnsaveUserName" --query "SSHPublicKeys[?Status == 'Active'].[SSHPublicKeyId]" --output text | while read -r KeyId; do
-  aws iam get-ssh-public-key --user-name "$UnsaveUserName" --ssh-public-key-id "$KeyId" --encoding SSH --query "SSHPublicKey.SSHPublicKeyBody" --output text
+raw_username="$1"
+raw_username=${raw_username//".plus."/"+"}
+raw_username=${raw_username//".equal."/"="}
+raw_username=${raw_username//".comma."/","}
+
+if [ "${STRIP_EMAILS_FROM_USERNAME}" -eq 1 ]; then
+    iam_username=$(aws iam list-users --query "Users[*].[UserName]" --output text | fgrep "$raw_username@")
```

Comment:
this will only get the first page of users. So with many IAM users, not all users will be returned on the first page.